File: mipush_hook/src/main/java/com/nihility/service/XMPushServiceListener.java
package com.nihility.service;

import android.content.Intent;

import com.xiaomi.push.service.XMPushService;

public interface XMPushServiceListener {

    default void initialize(XMPushService pushService) {
    }

    default void created() {
    }

    default void destroy() {
    }

    default void start(Intent intent) {
    }

    enum ConnectionStatus {
        connecting,
        connected,
        disconnected;

        public static ConnectionStatus of(int i) {
            return values()[i];
        }
    }

    default void connectionStatusChanged(ConnectionStatus connectionStatus) {
    }
}

File: xmsf/shell/src/main/java/io/github/magisk317/mipush/service/XMPushServiceListener.kt
package io.github.magisk317.mipush.service

import io.github.magisk317.mipush.runtime.core.ConnectionStatus

import android.content.Intent

interface XMPushServiceListener {
    fun created() {}
    fun destroy() {}
    fun start(intent: Intent) {}

    typealias ConnectionStatus = io.github.magisk317.mipush.runtime.core.ConnectionStatus

    fun connectionStatusChanged(connectionStatus: ConnectionStatus) {}
}

typealias ConnectionStatus = io.github.magisk317.mipush.runtime.core.ConnectionStatus
